File: Week 8 - Angular and APIs/IntroToFullstack/SchoolFrontend/SchoolFrontend/src/app/Student.ts
// To parse this data:
//
//   import { Convert, Student } from "./file";
//
//   const student = Convert.toStudent(json);

//Remember this model needs to match the spelling and casing of what the C# API has in its JSON 
export interface Student {
    id:           number;
    name:         string;
    age:          number;
    averageGrade: number;
    isPassing:    boolean;
}

// Converts JSON strings to/from your types
export class Convert {
    public static toStudent(json: string): Student {
        return JSON.parse(json);
    }

    public static studentToJson(value: Student): string {
        return JSON.stringify(value);
    }
}

File: Week 8 - Angular and APIs/IntroToFullstack/SchoolBackend/SchoolBackend/Models/Student.cs
using System;
using System.Collections.Generic;

namespace SchoolBackend.Models;

//This model needs to match the datatype, spelling and casing from the database 
public partial class Student
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public int? Age { get; set; }

    public double? AverageGrade { get; set; }

    public bool? IsPassing { get; set; }
}
